src/common/slow-query/src/recorder.rs

```diff
@@ -0,0 +1,355 @@
+                    query = query,
+                    is_promql = is_promql
+                );
+            }
+            SlowQueriesRecordType::SystemTable => {
+                // Record the slow query in a system table.
+                if let Err(e) = self
+                    .insert_slow_query(cost, threshold, &query, is_promql)
+                    .await
+                {
+                    error!(e; "Failed to insert slow query, query: {}", query);
+                }
+            }
+        }
+    }
+
+    async fn insert_slow_query(
+        &self,
+        cost: u64,
+        threshold: u64,
+        query: &str,
+        is_promql: bool,
+    ) -> Result<()> {
+        if let Some(table) = self
+            .catalog_manager
+            .table(
+                self.query_ctx.current_catalog(),
+                DEFAULT_PRIVATE_SCHEMA_NAME,
+                SLOW_QUERY_TABLE_NAME,
+                Some(&self.query_ctx),
+            )
+            .await
+            .context(CatalogSnafu)?
+        {
+            let insert = RowInsertRequest {
+                table_name: SLOW_QUERY_TABLE_NAME.to_string(),
+                rows: Some(Rows {
+                    schema: self.build_insert_column_schema(),
+                    rows: vec![Row {
+                        values: vec![
+                            ValueData::U64Value(cost).into(),
+                            ValueData::U64Value(threshold).into(),
+                            ValueData::StringValue(query.to_string()).into(),
+                            ValueData::BoolValue(is_promql).into(),
+                            ValueData::TimestampNanosecondValue(
+                                Timestamp::current_time(TimeUnit::Nanosecond).value(),
+                            )
+                            .into(),
+                        ],
+                    }],
+                }),
+            };
+
+            let requests = RowInsertRequests {
+                inserts: vec![insert],
+            };
+
+            let table_info = table.table_info();
+            let query_ctx = QueryContextBuilder::default()
+                .current_catalog(table_info.catalog_name.to_string())
+                .current_schema(table_info.schema_name.to_string())
+                .build()
+                .into();
+
+            self.inserter
+                .handle_row_inserts(requests, query_ctx, &self.statement_executor)
+                .await
+                .context(InsertRowsSnafu)?;
+
+            Ok(())
+        } else {
+            TableNotFoundSnafu {
+                catalog: self.query_ctx.current_catalog().to_string(),
+                schema: DEFAULT_PRIVATE_SCHEMA_NAME.to_string(),
+                table: SLOW_QUERY_TABLE_NAME.to_string(),
+            }
+            .fail()
+        }
+    }
+
+    async fn create_system_table(&self) -> Result<()> {
+        let mut create_table_expr = self.build_create_table_expr(self.query_ctx.current_catalog());
```

Comment:
expr creation is non-trivial, better to move it after table existence check, for performance consideration